### PR TITLE
Docs: Rename "api" to "lib"

### PR DESCRIPTION
**Pull request by AlessandroA**
This is currently required by the docs.mbed.com engine, which expects
Doxygen documentation to be placed in an `api` folder, and will not recognize
our Markdown files.

Once the website moves to a different engine, we can revert back to
"api", if we want to.

@iriark01 Could you please verify that this fixes the issue you reported?

@Patater

### CODE REVIEW

File: docs/core/PORTING.md [hunk 1]
```diff
@@ -280,7 +280,7 @@ $ cd ~/code/uvisor
 $ make
 ```
 
-The build process generates as many static libraries (`*.a` files) as your family configurations, multiplied by two (debug and release builds). You can find them in `~/code/uvisor/api/lib/${family}`. Please note that these libraries are not published in the uVisor repository. Instead, the glue-layer library deploys them and makes them available to the target OS.
+The build process generates as many static libraries (`*.a` files) as your family configurations, multiplied by two (debug and release builds). You can find them in `~/code/uvisor/lib/lib/${family}`. Please note that these libraries are not published in the uVisor repository. Instead, the glue-layer library deploys them and makes them available to the target OS.
```
Comment (iriark01) [inline docs/core/PORTING.md]: Did you mean to change this one?

File: docs/lib/manual/UseCases.md [hunk 2]
```diff
@@ -691,4 +691,4 @@ void example(void)
 ```
 
 ### More information about the RPC API
-For more information about the RPC API, please refer to [the well-commented RPC API C header file](https://github.com/ARMmbed/uvisor/blob/master/api/inc/rpc.h).
+For more information about the RPC API, please refer to [the well-commented RPC API C header file](https://github.com/ARMmbed/uvisor/blob/master/lib/inc/rpc.h).
```
Comment (iriark01) [inline docs/lib/manual/UseCases.md]: Did you mean to change this one?

File: docs/core/PORTING.md (hunk 1 above)
Comment (AlessandroA) [inline docs/core/PORTING.md]: Good catch! Fixed :)

File: docs/lib/manual/UseCases.md (hunk 2 above)
Comment (AlessandroA) [inline docs/lib/manual/UseCases.md]: Good catch! Fixed :)

**Comment by iriark01** [comment]
LGTM, but I will have to build once to find out (after you merge)

**Comment by AlessandroA** [comment]
The CI is red because of backwards-incompatibility. This PR is now blocked by https://github.com/ARMmbed/mbed-os/pull/4179.

**Comment by AlessandroA** [comment]
ARMmbed/mbed-os#4179 has been merged. I'm updating the tests so that this PR can go green as well.

**Comment by AlessandroA** [comment]
retest uvisor

**Pull request merged**